flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/RecoveredChannelStateHandler.java

```diff
@@ -88,38 +127,66 @@ private RecoveredInputChannel getChannel(InputChannelInfo info) {
 	private final ResultPartitionWriter[] writers;
 	private final boolean notifyAndBlockOnCompletion;
 
-	ResultSubpartitionRecoveredStateHandler(ResultPartitionWriter[] writers, boolean notifyAndBlockOnCompletion) {
+	private final InflightDataRescalingDescriptor channelMapping;
+
+	private final Map<ResultSubpartitionInfo, List<CheckpointedResultSubpartition>> rescaledChannels = new HashMap<>();
+
+	ResultSubpartitionRecoveredStateHandler(ResultPartitionWriter[] writers, boolean notifyAndBlockOnCompletion, InflightDataRescalingDescriptor channelMapping) {
 		this.writers = writers;
+		this.channelMapping = channelMapping;
 		this.notifyAndBlockOnCompletion = notifyAndBlockOnCompletion;
 	}
 
 	@Override
 	public BufferWithContext<Tuple2<BufferBuilder, BufferConsumer>> getBuffer(ResultSubpartitionInfo subpartitionInfo) throws IOException, InterruptedException {
-		BufferBuilder bufferBuilder = getSubpartition(subpartitionInfo).requestBufferBuilderBlocking();
+		final List<CheckpointedResultSubpartition> channels = getMappedChannels(subpartitionInfo);
+		BufferBuilder bufferBuilder = channels.get(0).requestBufferBuilderBlocking();
```

Comment:
nit: if `channels` is empty then a more informative error message would be helpful